### ISSUE TITLE
Make names of variables and actions different between the component and the controller.

### ISSUE THREAD

**Issue by joetidee** [emberjs/guides#1387]
From reading the tutorial, it was difficullt to discern which name related to either the component or the controller:

[https://guides.emberjs.com/v2.5.0/tutorial/autocomplete-component/](https://guides.emberjs.com/v2.5.0/tutorial/autocomplete-component/)

I had some assitance understanding this on the Slack forum, however, I think it would be advatangeous to newbies like myself if some naming adjustments could be made.

For example, here I understood that the component was being inserted into the template file:

`{{filter-listing filteredList=filteredList autoComplete=(action 'autoComplete') search=(action 'search')}}`

However, what wasn't clear was where the values where coming from because the flow of the explanation in the text was broken up by the text explaingin how to generate the controller file.

What would help, is to do something like this:

`{{filter-listing filteredList=filteredList controllerAutoComplete=(action 'controllerAutoComplete') controllerSearch=(action 'controllerSearch')}}`

> filter-listing.js:

```
export default Ember.Component.extend({
  filter: null,
  filteredList: null,
  actions: {
    componentAutoComplete() {
      this.get('controllerAutoComplete')(this.get('filter'));
    },
    componentSearch() {
      this.get('controllerSearch')(this.get('filter'));
    },
    componentChoose(city) {
      this.set('filter', city);
    }
  }
});
```

> /controllers/index.js:

```
export default Ember.Controller.extend({
  filteredList: null,
  actions: {
    controllerAutoComplete(param) {
      if (param !== '') {
        this.store.query('rental', { city: param }).then((result) => {
          this.set('filteredList', result);
        });
      } else {
        this.set('filteredList', null);
      }
    },
    controllerSearch(param) {
      if (param !== '') {
        this.store.query('rental', { city: param }).then((result) => {
          this.set('model', result);
        });
      } else {
        this.store.findAll('rental').then((result) => {
          this.set('model', result);
        });
      }
    }
  }
});
```

> /templates/components/filter-listing.js:

```
City: {{input value=filter key-up=(action 'componentAutoComplete')}}
<button {{action 'componentSearch'}}>Search</button>
<ul>
  {{#each filteredList as |item|}}
      <li {{action 'componentChoose' item.city}}>{{item.city}}</li>
  {{/each}}
</ul>
```

**Comment by acorncom** [comment]
@joetidee Those sound like good suggestions, would you be able to send in a PR with your changes?

**Comment by joetidee** [comment]
No idea how to do this.

**Comment by hankfanchiu** [comment]
@acorncom Wanted to help out. See my PR ^

**Comment by toddjordan** [comment]
Commented on the PR. Would prefer that the component not have knowledge of the controller and be agnostic on who its parent is.  There may be other ways to make this clearer, such as removing the actions from the component.

**Comment by hankfanchiu** [comment]
If this https://github.com/emberjs/super-rentals/pull/17 makes sense then I'll update the PR for the guide accordingly.

**Comment by toddjordan** [comment]
Closing as fixed by #1401 
Thanks!
